### ISSUE TITLE
daemon.php status says not running with many zombies process

### ISSUE THREAD

**Issue by valvin1** [friendica/docker#105]
Hi,

I've just seen that `exec php /var/www/html/bin/daemon.php -f start` is not terminated when `daemon.php status` return it is not running. Maybe it is normal. 

But we have strange issue on our instance which generates maybe zombies process : 

![image](https://user-images.githubusercontent.com/1267915/75608984-f4022700-5b04-11ea-95f2-ca2b9c32d5d3.png)

In that situation we have many messages queueing. 

if I run dameon.php in background inside the container i get the following result. Is it expected result ?

```
/var/www/html/bin# ./daemon.php status
Pidfile wasn't found. Is the daemon running?
/var/www/html/bin# ./daemon.php start 
Starting worker daemon.
/var/www/html/bin# ./daemon.php status
Daemon process 30620 is running.
```
Then messages unqueued.

Do you have any idea why worker looses their parent id ?

Thank you :)

**Comment by valvin1** [comment]
in addition on my own instance I'm checking zombie process and here the result : 

```
 ps -ef|grep php|grep defunct|wc -l
7065
```
:scream:

**Comment by nupplaphil** [comment]
Hmm .. I think this is a serious problem .. I have these zombie processes too!
So even the official suggested `cron.sh` container has this issue!

I think this is because of the way we use forking inside the `daemon.php` itself, which causes such zombie processes.. I've to investigate it!

Thanks for your input!

**Comment by valvin1** [comment]
interesting article. Maybe a first step would be to use this approach : 

```
FROM friendica/server
CMD ["/bin/bash", "-c", "set -e && /cron.sh"]
```

**Comment by valvin1** [comment]
if i remove `exec` in `cron.sh` and use the trick it looks working : 

![image](https://user-images.githubusercontent.com/1267915/75718243-e6020100-5cd2-11ea-9982-5e40d361a9d1.png)

left is with `exec` so child process are managed by `php` ; right they are managed by `sh` and no more defunct process.

**Comment by nupplaphil** [comment]
Yes!! Thx!! I haven't had time for this test :))
I'll create a PR for it

**Comment by valvin1** [comment]
I think that just remove `exec` in `cron.sh` can fix the issue.

**Comment by nupplaphil** [comment]
funny .. This was an explicit recommendation of the docker maintainer team:
https://github.com/docker-library/official-images/pull/6446#issuecomment-537282517 :D

**Comment by valvin1** [comment]
it makes sense here the issue is daemon.php is not managing its children :)
btw in current situation it would be better not having `exec`
I'm sending a PR with just removing exec.

**Comment by nupplaphil** [comment]
@valvin1 can you plz have a look at https://github.com/docker-library/official-images/pull/7573#issuecomment-594976003 and adapt the solution? This would help me a lot :))

**Comment by valvin1** [comment]
interesting i'll try to do it this week-end :)

**Comment by nupplaphil** [comment]
Fixed with #108

**Comment by nupplaphil** [comment]
btw. I checked it at both of my nodes, it's now working!! :-)
